app/models/notifier.rb

```diff
@@ -183,24 +180,42 @@ def changeset_comment_notification(comment, recipient)
                   I18n.t("notifier.changeset_comment_notification.commented.subject_other", :commenter => @commenter)
                 end
 
-      attach_user_avatar(comment.author)
-
-      mail :to => recipient.email, :subject => subject
+      compose_mail recipient,
+                   :subject => subject,
+                   :attach_avatar => comment.author
     end
   end
 
   private
 
+  def compose_mail(recipient, options = {})
+    needs_avatar = options.include? :attach_avatar
+    avatar_user = options.delete :attach_avatar
+    options[:to] ||= recipient.email
+    mail(options) do |format|
+      if recipient.preferred_email_format == "text_only"
+        format.text
+      else
+        unless recipient.preferred_email_format == "multipart"
+          logger.warn "Unknown email format '#{recipient.preferred_email_format}, treated as 'multipart'"
+        end
```

Comment:
Yes, this was more to guard against future changes to the code that wouldn't properly update this method, than against something which can occur in the code as it now is. I'm happy to remove it.